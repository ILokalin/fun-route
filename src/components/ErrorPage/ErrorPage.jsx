import React from 'react';
import { Link } from 'react-router-dom';


export default class extends React.Component {

  render () {

    return (
      <section className="error-page">
        <p className="error-page__number-404">Точка
          <span className="error-page__number-404--big"> 404</span></p>
        <header className="error-page__header">
          <h2 className="error-page__title">Внимание! Вы ушли с маршрута... </h2>
        </header>
        <p className="error-page__text">Вероятно, была выбрана неверная ссылка в адресной строке. Чтобы вернуться в приложение перейдите по ссылке:
          <Link className="error-page__link" to="/">Вернуться в приложение</Link>
        </p>
      </section>
    )
  }
}