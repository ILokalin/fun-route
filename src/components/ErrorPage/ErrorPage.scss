.error-page {
  max-width: 1160px;
  margin: 0 auto;
  color: $gray-lightnes;
  padding-left: $side-gap;
  padding-right: $side-gap;

  @media (min-width: $screen-max) {
    padding: 0;
  }


  &__header {
    text-align: center;
    padding-bottom: 6.25rem;
  }

  &__title {
    color: $color-secondary;
    opacity: 0.5;
  }


  &__text {
    margin: 0 auto;
    width: 18.75rem;
    line-height: 1.875rem;

    @media (min-width: $screen-sd) {
      width: 37.5rem;
    }
  }


  &__link {
    display: block;
    color: $color-secondary;
    transition: all 0.3s linear;
    

    &:hover {
      color: $color-main;
    }
  }


  &__number-404 {
    width: 100%;
    padding-top: 6.25rem;
    font-size: 3rem;
    margin: 0 auto;
    position: relative;
    text-align: center;
    color: $color-secondary;
    opacity: 0.5;
    padding-bottom: 2rem;

    &--big {
      font-size: 4rem;
    }
  }
}
