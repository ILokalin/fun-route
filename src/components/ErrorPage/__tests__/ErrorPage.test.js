import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import rendrer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import ErrorPage from 'components/ErrorPage';


describe('ErrorPage Component', () => {
  it('render snap', () => {
    const component = rendrer.create(
      <BrowserRouter>
        <ErrorPage />
      </BrowserRouter>
      );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    })
})
