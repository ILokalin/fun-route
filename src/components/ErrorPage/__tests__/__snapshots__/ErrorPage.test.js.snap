// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage Component render snap 1`] = `
<section
  className="error-page"
>
  <p
    className="error-page__number-404"
  >
    Точка
    <span
      className="error-page__number-404--big"
    >
       404
    </span>
  </p>
  <header
    className="error-page__header"
  >
    <h2
      className="error-page__title"
    >
      Внимание! Вы ушли с маршрута... 
    </h2>
  </header>
  <p
    className="error-page__text"
  >
    Вероятно, была выбрана неверная ссылка в адресной строке. Чтобы вернуться в приложение перейдите по ссылке:
    <a
      className="error-page__link"
      href="/"
      onClick={[Function]}
    >
      Вернуться в приложение
    </a>
  </p>
</section>
`;
